src/DataStructures/Tensor/Expressions/Contract.hpp

```diff
@@ -187,7 +187,7 @@ template <template <typename> class TE, typename ReplacedArgList, size_t I,
 SPECTRE_ALWAYS_INLINE static constexpr auto fully_contract(const T& t) {
   // TensorIndex types of pair of indices to contract
   using lower_tensorindex = ti_contracted_t<I, UpLo::Lo>;
-  using upper_tensorindex = TensorIndex<lower_tensorindex::value + 1, UpLo::Up>;
+  using upper_tensorindex = ti_contracted_t<I, UpLo::Up>;
```

Comment:
aha, now you found the bug :)

Reply:
See [the above comment](https://github.com/sxs-collaboration/spectre/pull/2584#discussion_r521122563)